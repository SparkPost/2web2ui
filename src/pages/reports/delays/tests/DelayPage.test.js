import React from 'react';
import { DelayPage } from '../DelayPage';
import { shallow } from 'enzyme';

jest.mock('src/helpers/reports');

describe('DelayPage: ', () => {
  const props = {
    reportOptions: { relativeRange: 'hour' },
    tableLoading: false,
    reasons: [
      {
        count_delayed: 100,
        count_delayed_first: 1,
        reason: 'my reason'
      }
    ],
    aggregates: {
      count_delayed: 1000,
      count_delayed_first: 10,
      count_accepted: 10000
    },
    totalAccepted: 1000,
    refreshDelayReport: jest.fn()
  };

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<DelayPage {...props} />);
  });

  it('should render', () => {
    expect(props.refreshDelayReport).not.toHaveBeenCalled();
    expect(wrapper).toMatchSnapshot();
  });

  it('should refresh when report options reference changes', () => {
    const newReportOptions = {};
    expect(props.refreshDelayReport).not.toHaveBeenCalled();
    wrapper.setProps({ reportOptions: newReportOptions });
    expect(props.refreshDelayReport).toHaveBeenCalledWith(newReportOptions);
  });

  it('should show loading indicator when loading table', () => {
    wrapper.setProps({ loading: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should show loading panel when aggregates are still loading', () => {
    wrapper.setProps({ aggregatesLoading: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should not display top level metrics when there are no aggregates', () => {
    wrapper.setProps({ aggregates: {}});
    expect(wrapper.find('MetricsSummary')).toHaveLength(0);
  });
});
