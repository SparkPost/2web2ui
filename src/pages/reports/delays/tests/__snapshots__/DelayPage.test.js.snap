// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DelayPage:  should render 1`] = `
<Page
  title="Delay Report"
>
  <withRouter(Connect(ReportOptions)) />
  <Connect(MetricsSummary)
    rateTitle="Delayed Rate"
    rateValue={0.1}
    secondaryMessage="There were 1,000 total delays in this time period. (Note: messages may be delayed multiple times)"
  >
    <strong>
      10
    </strong>
     of
     
    <strong>
      10,000
    </strong>
     accepted messages were delayed on first attempt
  </Connect(MetricsSummary)>
  <DelaysDataTable
    rows={
      Array [
        Object {
          "count_delayed": 100,
          "count_delayed_first": 1,
          "reason": "my reason",
        },
      ]
    }
    totalAccepted={1000}
  />
</Page>
`;

exports[`DelayPage:  should show loading indicator when loading table 1`] = `
<Page
  title="Delay Report"
>
  <withRouter(Connect(ReportOptions))
    reportLoading={true}
  />
  <Connect(MetricsSummary)
    rateTitle="Delayed Rate"
    rateValue={0.1}
    secondaryMessage="There were 1,000 total delays in this time period. (Note: messages may be delayed multiple times)"
  >
    <strong>
      10
    </strong>
     of
     
    <strong>
      10,000
    </strong>
     accepted messages were delayed on first attempt
  </Connect(MetricsSummary)>
  <PanelLoading
    accent={false}
    minHeight="400px"
  />
</Page>
`;

exports[`DelayPage:  should show loading panel when aggregates are still loading 1`] = `
<Page
  title="Delay Report"
>
  <withRouter(Connect(ReportOptions)) />
  <PanelLoading
    accent={false}
    minHeight="115px"
  />
  <DelaysDataTable
    rows={
      Array [
        Object {
          "count_delayed": 100,
          "count_delayed_first": 1,
          "reason": "my reason",
        },
      ]
    }
    totalAccepted={1000}
  />
</Page>
`;
